src/subclass/widget.rs

```diff
@@ -1349,3 +1368,17 @@ unsafe extern "C" fn widget_get_preferred_height_for_width<T: ObjectSubclass>(
         *nat_height_ptr = nat_height;
     }
 }
+
+unsafe extern "C" fn widget_size_allocate<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkWidget,
+    allocation: *mut gtk_sys::GtkAllocation,
+) where
+    T: WidgetImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Widget = from_glib_borrow(ptr);
+    let allocate: Allocation = from_glib_none(allocation);
```

Comment:
If the allocation should be possible to modify then you need to write it back afterwards.

Reply:
The allocation is not an in-out parameter; you're supposed to call `gtk_widge_set_allocation()` or chain up to the parent class's allocation, in which case you can tweak the argument for those.

Reply:
Thanks, so the allocation should actually be an immutable reference. And new allocations could be passed to childs, etc.